incubator/hnc/internal/validators/object.go

```diff
@@ -246,7 +246,19 @@ func (o *Object) hasConflict(inst *unstructured.Unstructured) (bool, []string) {
 	// Get a list of conflicting descendants if there's any.
 	for _, desc := range descs {
 		if o.Forest.Get(desc).HasSourceObject(gvk, nm) {
-			conflicts = append(conflicts, desc)
+			// check if the selectors match
+			selector, _ := selectors.GetSelector(inst)
+			treeSelector, _ := selectors.GetTreeSelector(inst)
+			noneSelector, _ := selectors.GetNoneSelector(inst)
+			nsLabels := o.Forest.Get(inst.GetNamespace()).GetLabels()
+			// We only check notMatch here because an empty selector annotation will be treated as an
+			// "all-match" selector
+			notMatch := (selector != nil && !selector.Matches(nsLabels)) ||
+				(treeSelector != nil && !treeSelector.Matches(nsLabels)) ||
+				noneSelector == true
+			if !notMatch {
```

Comment:
Seems it can be
```
match := (selector ==nil || selector.Matchers(l)) &&
 (treeSelector == nil || treeSelector.Matches(l)) &&
 noneSelector == false
```
so that you can use `if match {}` instead of `if !notMatch{}`?

Reply:
Ture, updated!